src/main/capi/duckdb_value-c.cpp

```diff
@@ -393,6 +393,91 @@ duckdb_value duckdb_create_array_value(duckdb_logical_type type, duckdb_value *v
 	return WrapValue(array_value);
 }
 
+duckdb_value duckdb_create_map_value(duckdb_logical_type map_type, duckdb_value keys, duckdb_value values) {
+	if (!map_type || !keys || !values) {
+		return nullptr;
+	}
+
+	auto &map_logical_type = UnwrapType(map_type);
+	if (duckdb::TypeVisitor::Contains(map_logical_type, duckdb::LogicalTypeId::INVALID) ||
+	    duckdb::TypeVisitor::Contains(map_logical_type, duckdb::LogicalTypeId::ANY)) {
+		return nullptr;
+	}
+
+	auto &key_logical_type = duckdb::MapType::KeyType(map_logical_type);
```

Comment:
This is assuming `map_logical_type` is a MAP, but this isn't verified anywhere before this, I assume there is no test for this either because this will throw an InternalException I'm pretty sure, can we add a test and fix this?